### PR TITLE
major refactoring of django views

### PR DESCRIPTION
**Pull request by apkallum**
# Summary

New public index derived from Django db.

# Changes these areas

- [ ] Bugfixes
- [x] Feature behavior
- [ ] Command line interface
- [ ] Configuration options
- [ ] Internal architecture
- [ ] Archived data layout on disk

# Roadmap Goals

Going to 0.5 and deriving all indexes from Django db.

### CODE REVIEW

File: archivebox/core/urls.py [hunk 1]
```diff
@@ -31,4 +31,5 @@
     path('index.html', RedirectView.as_view(url='/')),
     path('index.json', static.serve, {'document_root': settings.OUTPUT_DIR, 'path': 'index.json'}),
     path('', MainIndex.as_view(), name='Home'),
+    path('public/', PublicArchiveView.as_view())
```
Comment (pirate) [inline archivebox/core/urls.py:34]: why a separate URL, cant this just be the default homepage when PUBLIC_INDEX=True?

Comment (apkallum) [reply archivebox/core/urls.py:34]: Would you like this view to replace the OldHome then?

File: archivebox/themes/default/core/snapshot_list.html [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                    iDisplayLength: 100,
+                });
+            });
+        </script>
+        <base href="{% url 'Home' %}">
+    </head>
+    <body data-status="finished">
+        <header>
+            <div class="header-top container-fluid">
+                <div class="row nav">
```
Comment (cdvv7788) [inline archivebox/themes/default/core/snapshot_list.html:219]: This navbar is not present in the `add` view. Given that all of the web interface will focus on django now, we can create actual django templates with a proper structure, having a parent template with the base structure, and overriding the content block with the new snapshot list and add form respectively.

File: archivebox/core/views.py [hunk 3]
```diff
@@ -102,3 +92,61 @@ def get(self, request, path):
...
+    extra_context = {'title': 'Add URLs'}
+
+    def get(self, request, *args, **kwargs):
+        if PUBLIC_ADD_VIEW or self.request.user.is_authenticated:
+            self.extra_context['form'] = AddLinkForm()
+            return render(template_name='add_links.html', request=request, context=self.extra_context)
+        else:
+            return redirect(f'/admin/login/?next={request.path}')
+
+    def post(self, request, *args, **kwargs):
```
Comment (cdvv7788) [inline archivebox/core/views.py:128]: Missing check for permissions in POST.

**Pull request closed without merging**